test/common/redis/conn_pool_impl_test.cc

```diff
@@ -65,9 +66,12 @@ class RedisClientImplTest : public testing::Test, public DecoderFactory {
         .WillOnce(SaveArg<0>(&upstream_read_filter_));
     EXPECT_CALL(*upstream_connection_, connect());
     EXPECT_CALL(*upstream_connection_, noDelay(true));
+
     client_ = ClientImpl::create(host_, dispatcher_, EncoderPtr{encoder_}, *this, *config_);
     EXPECT_EQ(1UL, host_->cluster_.stats_.upstream_cx_total_.value());
     EXPECT_EQ(1UL, host_->stats_.cx_total_.value());
+
+    ON_CALL(*host_, outlierDetector()).WillByDefault(ReturnRef(outlier_detector_));
```

Comment:
Can you move the mock and the ON_CALL into the mock host? That way others can benefit in the future and avoid duplication.